### ISSUE TITLE
Does Flysky OEM 2022 Firmware include you bug fix?

### ISSUE THREAD

**Issue by Growflavor** [Cleric-K/FlySky-i6X_Firmware_BugFix#16]
Thank you for this informative github repository & responses in issue #12 

Would be great to know if FlySky finally included your fix in the 2022/5/18 firmware v. 1.0.52.1442

https://www.flysky-cn.com/i6x-xiazai-1

**Comment by Cleric-K** [comment]
Hi, thanks for pointing out this new update. I didn't know about it thus I have no idea if it has been fixed. I'll try it when I can but it might take few weeks since I have some other more pressing things at this time. Whether it has been fixed can be seen through some blackbox logging where the raw rc commands are recorded.

**Comment by Growflavor** [comment]
RE: "Whether it has been fixed can be seen through some blackbox logging "  I am not yet that far along in my skill working with these controllers ;-)  Thus, if & when you ever might have time to look into it, would be useful to know.

Also, fyi...the OEM updater for the 2022 firmware is not flagged as malicious which is nice.  

Just to tinker, I tried to pack the 2019 fixed .bin you provided into the 2022 updater using resource hacker, but seems (at least with me not knowing what I was doing) the new exe file that resource hacker creates is identified by malware tools as a 'trojan' rather than as a new unique .exe...but again I am not familiar with this tool so probably did many things incorrectly.

**Comment by joewolf93** [comment]
Hi! I know it's an already old post, but can you write us some update please? I bought my FS-i6x right now and would like to update firmware. Thanks!

**Comment by Cleric-K** [comment]
Sorry for taking so long (I've been away from RC for quite some time). The 2022 version still has the bug. You can find the patched version in the releases.

**Comment by joewolf93** [comment]
Thank you for your response. Unfortunately I can not install you version because after the install everything goes off screen

**Comment by Cleric-K** [comment]
Strange. I have tested it and it seems to work fine here. Were you already with the official 2022 update?

**Comment by joewolf93** [comment]
Yes, I connected my device to the official app and updated my system. After that I tried flashing your release, but when I started the Tx it only showed a few pixels of text on the bottom of the screen. I could flash back the official firmware and it was fine after that. The problem is I can sense the lag produced by the official firmware even without black box testing.

**Comment by Cleric-K** [comment]
The only thing I can think of is that newer versions of the TX have new bootloader code which does some kind of checksum verification. This might be solvable but some experimentation would be needed. First we'll have to try flash a firmware with minimal change (only one byte difference) which doesn't affect the code. For example, a letter in some of the messages. If the firmware fails to run we can be pretty confident that it fails a checksum.

To solve this, however, I'll need a dump of the full memory of the TX, which can only be obtained by wiring a ST-LINK to the board. If you are willing to experiment, we can try.
